### DISCUSSION TITLE
Potential memory leak around ActivePartExpression, TrackableComputationExt and EclipseContext, detected with Sirius

### DISCUSSION THREAD

**Discussion by lredor** [eclipse-platform/eclipse.platform#889] [Q&A]
Hi,
I have made some analysis about memory leaks in [Sirius](https://eclipse.dev/sirius/). One of them concerns `ActivePartExpression`, `TrackableComputationExt` and `EclipseContext`. This issue is [#178-Memory leak through EclipseContext and ActivatePartExpression](https://github.com/eclipse-sirius/sirius-desktop/issues/178).

I have made some analyses (through debug) but I not really understand what is wrong and what is supposed to be done. Can someone, who knows these mechanisms, take a look at my [analyses](https://github.com/eclipse-sirius/sirius-desktop/issues/178#issuecomment-1829445154)?

I'm interested in any leads that might help me resolve the issue.

Thanks in advance,

Laurent

**Comment by lredor** [comment]
I maybe have a fix, for my memory leak problem. What do you think about ?
```
diff --git a/bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/contexts/ContextService.java b/bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/contexts/ContextService.java
index 2936de7..b2ace69 100644
--- a/bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/contexts/ContextService.java
+++ b/bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/contexts/ContextService.java
@@ -180,6 +180,9 @@
 			final UpdateExpression rat = activationToRat.remove(activation);
 			if (rat != null) {
 				rat.updating = false;
+				// The updating status has been set to false, so the expression is no longer
+				// useful. Set it to null to avoid potential memory leak.
+				rat.expression = null;
 				if (rat.cached != null && rat.cached != EvaluationResult.FALSE) {
 					contextService.deactivateContext(activation.getContextId());
 				}
```

**Reply by mickaelistria** [reply]
Please create a PR for it to facilitate discussion (and get tests running with this change).

**Reply by lredor** [reply]
OK. I'm currently on branch "R4_29_maintenance" (branch used for my memory leak analysis). Is it better to push the PR on master orcan I do it on branch that I actually used ?

**Reply by mickaelistria** [reply]
Please use master whenever possible.

**Reply by lredor** [reply]
Done: https://github.com/eclipse-platform/eclipse.platform.ui/pull/1341
